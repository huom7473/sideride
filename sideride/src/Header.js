import logo from "./logo.svg";
import React from "react";
import Amplify from 'aws-amplify';
import { Auth } from 'aws-amplify';
import awsconfig from './aws-exports';
import styled from 'styled-components';
import { Button } from "react-bootstrap";
import { useHistory } from "react-router";

Amplify.configure(awsconfig);




const Header = () => {
    const history = useHistory();

    const redirectSearch = () => {
        history.push('/');
    }

    const redirectCreateRide = () => {
        history.push('/createride');
    }

    const signOut = () => {
        try {
            Auth.signOut().then((res) => { history.push('/'); window.location.reload(false) });
        } catch (error) {
            console.log('error signing out: ', error);
        }
    }

    return (
        <header className="App-header">
            <img src={logo} className="App-logo" alt="logo" />
            <SideBarContainer>
                <Button onClick={redirectCreateRide}>Host Ride</Button>
                <Button onClick={redirectSearch}>Search Rides</Button>
                <Button>My Rides</Button>
                <Button onClick={signOut}>Sign Out</Button>
            </SideBarContainer>
        </header>
    )
}

const SideBarContainer = styled.div`
    width:500px;
    display:flex;
    flex-direction:row;
    justify-content:space-around;
    text-align:center;
`;

export default Header;