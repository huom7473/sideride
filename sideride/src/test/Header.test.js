import Header from "../Header"
import { configure } from 'enzyme';
import renderer from 'react-test-renderer';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import Amplify from 'aws-amplify';
import awsconfig from '../aws-exports';


Amplify.configure(awsconfig);
// Auth.signIn("test_user", "test_password");

configure({ adapter: new Adapter() });      // for enzyme methods like mount

test('Header matches snapshot', () => {
  const tree = renderer.create(<Header />).toJSON();
  expect(tree).toMatchSnapshot();
});

