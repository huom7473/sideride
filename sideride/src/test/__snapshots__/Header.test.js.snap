// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header matches snapshot 1`] = `
<header
  className="App-header"
>
  <img
    alt="logo"
    className="App-logo"
    src="logo.svg"
  />
  <div
    className="sc-bdvvtL drbwPJ"
  >
    <button
      className="btn btn-primary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Host Ride
    </button>
    <button
      className="btn btn-primary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Search Rides
    </button>
    <button
      className="btn btn-primary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      My Rides
    </button>
    <button
      className="btn btn-primary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Sign Out
    </button>
  </div>
</header>
`;
